test/unit/connection/result/result_test_duplicated_columns.js

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (c) 2015-2019 Snowflake Computing Inc. All rights reserved.
+ */
+
+const assert = require('assert');
+const ResultTestCommon = require('./result_test_common');
+const RowMode = require('./../../../../lib/constants/row_mode');
+
+describe('Result: renamed columns depend on row mode', function () {
+
+  const columnNames = ['KEY', 'FOO', 'KEY_1', 'FOO', 'KEY_3', 'FOO_3', 'KEY', 'FOO', 'KEY', 'FOO'];
+  const testCases = [
+    {
+      title: 'should return renamed columns for duplicates if ro mode object_with_renamed_duplicated_columns',
+      rowMode: RowMode.OBJECT_WITH_RENAMED_DUPLICATED_COLUMNS,
+      expectedColumnNames: [
+        'KEY',
+        'FOO',
+        'KEY_1',
+        'FOO_2',
+        'KEY_3',
+        'FOO_3',
+        'KEY_2',
+        'FOO_4',
+        'KEY_4',
+        'FOO_5'
+      ]
+    },
+    {
+      title: 'should not rename if row mode object',
+      rowMode: RowMode.OBJECT,
+      expectedColumnNames: columnNames
+    },
+    {
+      title: 'should not rename if row mode array',
+      rowMode: RowMode.ARRAY,
+      expectedColumnNames: columnNames
+    }
+  ];
+  const responseWithColumns = (columnRowSet) => {
+    return {
+      'data': {
+        'parameters': [{'name': 'CLIENT_PREFETCH_THREADS', 'value': 4}, {
```

Comment:
do we need all those parameters?

Reply:
I've check edthat test working without parameters too.